swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -445,25 +393,18 @@ extension Adapter: CallbackHandlerDelegate {
   }
 
   public func onDisconnect(error: String) {
+    // Immediately invalidate our session pointer to prevent workQueue items from trying to use it
+    session = nil
```

Comment:
Just to be sure we don't leak memory, can you insert a call to `disconnect` here? Otherwise I think you'd have to free the memory.

Reply:
Oh, is that the case? So connlib calls our `on_disconnect` and then expects us to issue a final `disconnect()` afterward?

Reply:
We'll also need to update the Android client if so

Reply:
👍